src/Management/src/EndpointBase/Metrics/Observer/EventCounterListener.cs

```diff
@@ -21,17 +21,25 @@ public class EventCounterListener : EventListener
     private readonly string _eventSourceName = "System.Runtime";
     private readonly string _eventName = "EventCounters";
     private readonly IMetricsObserverOptions _options;
+    private readonly bool _isInitialized = false;
```

Comment:
I think this can be removed. It is always set to `true` from the constructor, and there are no other constructors, so it will never be `false`.

Reply:
There is a wierd case of of timing because of how event listener subscribers work. So this code is needed because the event is fired before the constructor is run.

Reply:
Can you elaborate on that? As far as I'm aware, its impossible in C# to change the value of a readonly field.

Reply:
So the timing as can be observed from this class, 

1.  the fields are initialized 
2. OnEventWritten(EventWrittenEventArgs eventData) gets invoked. ( at this time , is initialized is false, and the events are added to a bag) 
3. Constructor runs (isInitialized is true) - bag is processed
4. OnEventWritten fires again and this time isInitialized is true, and bag is not updated and the event is processed.

Reply:
We discussed this during a meeting. I suspect the base class constructor invokes `OnEventWritten`, at which point the derived constructor hasn't run. Either way, the code works as intended, so I'm closing this.